crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -799,72 +799,93 @@ impl<'a> TimelineEventHandler<'a> {
                         trace!(idx, "Replacing existing event");
                         self.items.set(idx, Arc::new(item.into()));
                         return;
-                    } else {
-                        // In more complex cases, remove the item and day
-                        // divider (if necessary) before re-adding the item.
-                        trace!("Removing local echo or duplicate timeline item");
-                        self.items.remove(idx);
-
-                        assert_ne!(
-                            idx, 0,
-                            "there is never an event item at index 0 because \
+                    }
+
+                    // In more complex cases, remove the item and day
+                    // divider (if necessary) before re-adding the item.
+                    trace!("Removing local echo or duplicate timeline item");
+                    self.items.remove(idx);
+
+                    assert_ne!(
+                        idx, 0,
+                        "there is never an event item at index 0 because \
                              the first event item is preceded by a day divider"
-                        );
+                    );
 
-                        // Pre-requisites for removing the day divider:
-                        // 1. there is one preceding the old item at all
-                        if self.items[idx - 1].is_day_divider()
+                    // Pre-requisites for removing the day divider:
+                    // 1. there is one preceding the old item at all
+                    if self.items[idx - 1].is_day_divider()
                             // 2. the item after the old one that was removed
                             //    is virtual (it should be impossible for this
                             //    to be a read marker)
                             && self
                                 .items
                                 .get(idx)
                                 .map_or(true, |item| item.is_virtual())
```

Comment:
I guess the comment is confusing rustfmt here, the indentation of these linse should be reduced.